File: server/src/main/scala/kpn/shared/NodeInfo.scala
package kpn.shared

import kpn.shared.common.ToStringBuilder
import kpn.shared.data.Tagable
import kpn.shared.data.Tags

case class NodeInfo(
  id: Long,
  active: Boolean,
  orphan: Boolean,
  country: Option[Country],
  name: String,
  names: Seq[NodeName],
  latitude: String,
  longitude: String,
  lastUpdated: Timestamp,
  tags: Tags,
  facts: Seq[Fact],
  location: Option[Location]
) extends Tagable with LatLon {

  def name(networkType: String): String = {
    tags(networkType + "_ref").getOrElse("no-name")
  }

  def name(networkType: NetworkType): String = {
    tags(networkType.nodeTagKey).getOrElse("no-name")
  }

  override def toString: String = ToStringBuilder(this.getClass.getSimpleName).
    field("id", id).
    field("active", active).
    field("orphan", orphan).
    field("country", country).
    field("name", name).
    field("names", names).
    field("latitude", latitude).
    field("longitude", longitude).
    field("lastUpdated", lastUpdated).
    field("tags", tags).
    field("facts", facts).
    build
}

File: spikes/ngrx/src/app/api/node-info.ts
import {NodeName} from './node-name';
import {Tag} from './tag';
import {Country} from "./country";

export interface NodeInfo {
  readonly id: number;
  readonly active: boolean;
  readonly orphan: boolean;
  readonly country: Country;
  readonly name: string;
  readonly names: NodeName[];
  readonly latitude: string;
  readonly longitude: string;
  readonly lastUpdated: string;
  readonly lastSurvey: string;
  readonly tags: Tag[];
  readonly facts: string[];
  readonly location: Location;
  readonly tiles: string[];
}
